addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.roku.internal.handler;
+
+import static org.openhab.binding.roku.RokuBindingConstants.*;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.roku.internal.RokuState;
+import org.openhab.binding.roku.internal.protocol.RokuCommands;
+import org.openhab.binding.roku.internal.protocol.RokuCommunication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RokuHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jarod Peters - Initial contribution
+ * @auther Shawn Wilsher - Overhaul of channels and properties
+ */
+public class RokuHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(RokuHandler.class);
+
+    public RokuHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        if (command.toFullString().equals("REFRESH")) {
+            updateData();
+        } else {
+            RokuCommands rokuDevice = new RokuCommands(ipAddress, port);
+            try {
+                rokuDevice.generateAction(channelUID);
+            } catch (IOException e) {
+                logger.error("IOException occurred when attempting action", e);
+            }
+            updateState(channelUID, OnOffType.OFF);
```

Comment:
Is OFF always the expected the response here, could you explain?

Reply:
I've added a comment to clarify this.